### PR TITLE
Serialization

### PR DESCRIPTION
**Pull request by mustafaiman**
Default serializers ,IdentifiedDataSerializable, custom serializer support and global serializer support added.

### CODE REVIEW

File: src/serialization/SerializationService.ts [hunk 1]
```diff
@@ -1,27 +1,251 @@
...
-        buffer.writeInt32BE(0, 0); // partition hash
-        buffer.writeInt32BE(-11, 4); //string serializer
-        buffer.writeInt32BE(jsonString.length, 8);
-        buffer.write(jsonString, 12);
+export class SerializationServiceV1 implements SerializationService {
+
+    private registry: {[id: number]: Serializer};
+    private serializerNameToId: {[name: string]: number};
+    private numberType: string;
+    private serialiationConfig: SerializationConfig;
```
Comment (GhostInAMachine) [inline src/serialization/SerializationService.ts]: Typo

File: test/serialization/PortableTest.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+var Config = require('../../.').Config;
+var SerializationService = require('../../lib/serialization/SerializationService');
+var expect = require('chai').expect;
+describe.skip('Portable Serialization', function() {
```
Comment (GhostInAMachine) [inline test/serialization/PortableTest.js]: Portable is not implemented yet, is it?

Comment (mustafaiman) [inline test/serialization/PortableTest.js]: No it is not. I will implement it in another PR because this PR is already big.

Comment (GhostInAMachine) [inline test/serialization/PortableTest.js]: Would you remove the file then?

Comment (GhostInAMachine) [inline test/serialization/PortableTest.js]: Nevermind, I now see that it is skipped

**Pull request merged**